cli/app.go

```diff
@@ -1583,6 +1589,81 @@ Example:
 				},
 			},
 		},
+		{
+			Name:  "auth-app",
+			Usage: "manage third party auth applications",
+			Subcommands: []*cli.Command{
+				{
+					Name:  "create",
+					Usage: "create a third party auth application",
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     generalFlagOrgID,
+							Usage:    "organization ID that will be tied to auth application",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     authApplicationFlagName,
+							Usage:    "name for the auth application",
+							Required: true,
+						},
+						&cli.StringSliceFlag{
+							Name:     authApplicationFlagOriginURIs,
+							Usage:    "origin uris for the auth application",
+							Required: false,
+						},
+						&cli.StringSliceFlag{
+							Name:     authApplicationFlagRedirectURIs,
+							Usage:    "redirect uris for the auth application",
+							Required: false,
+						},
+						&cli.StringFlag{
+							Name:     authApplicationFlagLogoutURI,
+							Usage:    "logout uri for the auth application",
+							Required: false,
+						},
+					},
+					Action: RegisterAuthApplicationAction,
+				},
+				{
+					Name:  "update",
+					Usage: "update a third party auth application",
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     generalFlagOrgID,
+							Required: true,
+							Usage:    "organization ID that will be tied to auth application",
+						},
+						&cli.StringFlag{
+							Name:     authApplicationFlagApplicationID,
+							Usage:    "id for the auth application",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     authApplicationFlagName,
+							Usage:    "name for the auth application. omitting this will overwrite the existing value.",
+							Required: true,
+						},
+						&cli.StringSliceFlag{
+							Name:     authApplicationFlagOriginURIs,
+							Usage:    "origin uris for the auth application. omitting this will overwrite the existing value.",
```

Comment:
yeah so annoyingly if you omit these fields, it sets them to empty so people need to reinput everything

Reply:
I thought you could use "PATCH" method to just update specific fields? From the fusionauth website:

```
You must specify all of the properties of the Application when calling this API with the PUT HTTP method. 
When used with PUT, this API doesn’t merge the existing Application and your new data. 
It replaces the existing Application with your new data.

Utilize the PATCH HTTP method to send specific changes to merge into an existing Application.
```

Otherwise any custom variables that are set directly via fusionauth will be re-written after an update for one of these fields

Reply:
oh i didnt see that option in the fusion auth golang client but let me look again

Reply:
ah yes i missed it its a different api called `PatchApplicationWithContext`. will switch over to that app side! thanks roxy

Reply:
addressing missing fields app side in separate PR: https://github.com/viamrobotics/app/pull/4664